staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -233,6 +233,18 @@ type Config struct {
 
 	// StorageVersionManager holds the storage versions of the API resources installed by this server.
 	StorageVersionManager storageversion.Manager
+
+	// DelayServerShutdown dictates when to initiate shutdown of the HTTP Server
+	// during the graceful termination of the apiserver.
+	// If true, we wait for existing requests in flight to be completed and
+	// then initiate a shutdown of the HTTP Server.
+	// If false, we initiate a shutdown of the HTTP Server
+	// as soon as ShutdownDelayDuration has elapsed.
+	DelayServerShutdown bool
+
+	// GracefulShutdown provides access to the various shutdown events
+	// that happen during the graceful termination of the apiserver.
+	GracefulShutdown GracefulShutdown
```

Comment:
is this nil before shutdown?

Reply:
wondering that this thing is and how to name it better. It is not "a shutdown".

Reply:
it is not `nil`, we create a new instance of it inside `config.Complete`. we need it instanced up front so we can use the events in handler chain build function.

Reply:
> wondering what this thing is and how to name it better. It is not "a shutdown".

`GracefulShutdownCoordinator`, `GracefulShutdownDriver`? let me know if you have any suggestions :)

Reply:
addressed in the refactor pr